File: relationship-manager/conversation-service/src/main/java/com/bank/rm/conversation/dto/ConversationDtos.java
package com.bank.rm.conversation.dto;

import com.bank.rm.common.dto.*;
import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.UUID;

public class ConversationDtos {

    public record StartConversationRequest(CommunicationChannel channel, UUID customerId, UUID anonymousSessionId) {}

    public record SendMessageRequest(String content) {}

    public record ConversationResponse(UUID conversationId, ConversationPhase currentPhase,
                                        ConversationStatus status, CommunicationChannel channel,
                                        Instant createdAt, Instant lastMessageAt) {}

    public record MessageResponse(UUID messageId, UUID conversationId, SenderType senderType,
                                   String content, ContentType contentType, ConversationPhase phase,
                                   Map<String, Object> entitiesExtracted, Instant createdAt) {}

    public record SendMessageResponse(MessageResponse customerMessage, MessageResponse aiResponse) {}

    public record ConversationDetailResponse(UUID conversationId, ConversationPhase currentPhase,
                                              ConversationStatus status, List<MessageResponse> messages,
                                              Map<String, Object> extractedData) {}

    private ConversationDtos() {}
}

File: relationship-manager/conversation-service/src/main/kotlin/com/bank/rm/conversation/dto/ConversationDtos.kt
package com.bank.rm.conversation.dto

import com.bank.rm.common.dto.*
import java.util.UUID

data class StartConversationRequest(
    val customerId: UUID? = null,
    val anonymousSessionId: UUID? = null,
    val channel: CommunicationChannel = CommunicationChannel.WEB
)

data class ConversationResponse(
    val conversationId: UUID,
    val currentPhase: ConversationPhase,
    val status: ConversationStatus,
    val channel: CommunicationChannel,
    val createdAt: String,
    val lastMessageAt: String
)

data class SendMessageRequest(
    val content: String,
    val contentType: ContentType = ContentType.TEXT
)

data class MessageResponse(
    val messageId: UUID,
    val conversationId: UUID,
    val senderType: SenderType,
    val content: String,
    val contentType: ContentType,
    val phase: ConversationPhase?,
    val entitiesExtracted: Map<String, Any>?,
    val createdAt: String
)

data class ConversationHistoryResponse(
    val conversationId: UUID,
    val currentPhase: ConversationPhase,
    val status: ConversationStatus,
    val messages: List<MessageResponse>,
    val extractedData: Map<String, Any>
)

data class HandoffRequest(
    val conversationId: UUID,
    val reason: String? = null
)

data class ConversationSummaryResponse(
    val conversationId: UUID,
    val summary: String,
    val phasesCompleted: List<ConversationPhase>,
    val extractedData: Map<String, Any>
)
